go-controller/pkg/config/config.go

```diff
@@ -189,11 +190,29 @@ func overrideFields(dst, src interface{}) {
 		panic("mismatched struct types")
 	}
 
-	for i := 0; i < dstStruct.NumField(); i++ {
-		dstField := dstStruct.Field(i)
-		srcField := srcStruct.Field(i)
+	// Iterate over each field in dst/src Type so we can get the tags,
+	// and use the field name to retrieve the field's actual value from
+	// the dst/src instance
+	var handled bool
+	dstType := reflect.TypeOf(dst).Elem()
+	for i := 0; i < dstType.NumField(); i++ {
+		structField := dstType.Field(i)
+		// Ignore private internal fields; we only care about overriding
+		// 'gcfg' tagged fields read from CLI or the config file
+		if _, ok := structField.Tag.Lookup("gcfg"); !ok {
+			continue
+		}
+		handled = true
+
+		dstField := dstStruct.FieldByName(structField.Name)
+		srcField := srcStruct.FieldByName(structField.Name)
+		if !dstField.IsValid() || !srcField.IsValid() {
+			panic(fmt.Sprintf("invalid struct %q field %q",
```

Comment:
Can we ever get here? If we get here, do we always get here after pod/container restart?

Reply:
@pecameron if we get here, it's a pretty serious code failure on our part, and we should panic.

Reply:
RE SSL, it's always been supported and yes we should switch over. But the problem in OpenShift land has always been how we handle creating the certificates that ovn-kubernetes should use. That's not a problem we can solve here upstream in ovn-kubernetes.